lib/franklin/posts/commands/create_post.ex

```diff
@@ -1,9 +1,84 @@
 defmodule Franklin.Posts.Commands.CreatePost do
-  defstruct [
-    :published_at,
-    :title,
-    :uuid
-  ]
+  use Ecto.Schema
 
-  # TODO: We might want to make a more explcit `new/1` constructor so that we can enforce turncation of published at at the command level since that is the max resolution we are insterested in storing.
+  import Ecto.Changeset
+  import Franklin.Posts.Validations
+
+  @type t :: %__MODULE__{
+          id: Ecto.UUID.t(),
+          published_at: DateTime.t(),
+          title: String.t()
+        }
+
+  # TODO: Add typedoc.
+
+  @type new_attrs :: %{
+          optional(:id) => Ecto.UUID.t(),
+          required(:published_at) => DateTime.t(),
+          required(:title) => String.t()
+        }
+
+  @type error_list :: %{atom() => list(String.t())}
+
+  @primary_key {:id, Ecto.UUID, autogenerate: false}
+
+  embedded_schema do
+    field :published_at, :utc_datetime
+    field :title, :string
+  end
+
+  @doc """
+  Given a map of potential values attempts to return a valid `CreatePost` command.
+
+  If the values are valid, returns: `{:ok, command}`.
+
+  If the values are invalid, returns: `{:error, list}` where `list` is a keyword
+  list breakdown of the invalid reasons.
+
+  ## Expected Keys:
+
+    * `title` - A string value between 3 and 100 characters in length.
+    * `published_at` - A `DateTime` value.
+    * `id` - (optional) An `Ecto.UUID` value that will be used as the identity
+      of this post. Will be generated if not provided.
+
+  ## Examples:
+
+     > iex> CreatePost.new(123, nil, nil)
+     > %{
+     >   id: ["is invalid"],
+     >   published_at: ["can't be blank"],
+     >   title: ["can't be blank"]
+     > }
+
+  """
```

Comment:
I'm still debating how helpful it is to document specific domain attributes on a command like this since those would also have to be repeated in all the other command docs. 

Going to leave this for now, but in future work, I am considering focusing on documenting the primary domain contexts in full, and these specific command attributes less so.